index.js

```diff
@@ -1,18 +1,23 @@
-var bufferEqual = require('buffer-equal-constant-time');
-var Buffer = require('safe-buffer').Buffer;
+var Buffer = require('buffer').Buffer;
 var crypto = require('crypto');
 var formatEcdsa = require('ecdsa-sig-formatter');
```

Comment:
Node crypto's own `dsaEncoding` option valued `ieee-p1363` can be used for ECDSA to completely remove the `ecdsa-sig-formatter` dependency. This was added in https://github.com/nodejs/node/pull/29292 and is available in all nodejs versions above v13.2.0 and v12.16.0.

Reply:
Hmm, can you show a code example of how to solve it...
maybe just do a little change and then i can just apply the changes ;)
````
```diff
+
-
```
````

Reply:
You would need a separate signer and verifier functions (like it is done for RSA-PSS) and instead of `padding` and `saltLength` options use the `dsaEncoding` option. Nothing special really.

- https://github.com/nodejs/node/blob/6a489df73b0ab64f8fab016a7a71e209066fdb82/test/parallel/test-crypto-sign-verify.js#L554-L557
- https://github.com/nodejs/node/blob/6a489df73b0ab64f8fab016a7a71e209066fdb82/test/parallel/test-crypto-sign-verify.js#L516C38-L516C38

The one-shot APIs are self-explanatory and documented

- https://nodejs.org/api/crypto.html#cryptosignalgorithm-data-key-callback
- https://nodejs.org/api/crypto.html#cryptoverifyalgorithm-data-key-signature-callback

Reply:
Sry, i'm too lost right now. don't work much with cryptografy. will leave this fully up to you guys to solve.

Reply:
It's really just a lift and shift from the old api to the new and for ecdsa having a dedicated signer/verifier with the dsaEncoding option.